modules/xcursor.nix

```diff
@@ -0,0 +1,75 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.xcursor;
+
+  cursorType = types.submodule {
+    options = {
+      package = mkOption {
+        type = types.package;
+        example = literalExample "pkgs.vanilla-dmz";
+        description = "Package providing the cursor theme.";
+      };
+
+      name = mkOption {
+        type = types.str;
+        example = "Vanilla-DMZ";
+        description = "The cursor name within the package.";
+      };
+
+      size = mkOption {
+        type = types.int;
+        default = 32;
+        example = 64;
+        description = "The cursor size.";
+      };
+    };
+  };
+
+in
+
+{
+  meta.maintainers = [ maintainers.league ];
+
+  options = {
+    xcursor = mkOption {
+      type = types.nullOr cursorType;
+      default = null;
+      description = ''
+        The X cursor theme and settings. There package pkgs.xorg.xcursorthemes
```

Comment:
I'd change `There package pkgs.xorg.xcursorthemes` to `The package <varname>xorg.xcursorthemes</varname>`. I.e., using removing the `pkgs.` and using docbook tags to indicate that this is a variable name.